### ISSUE TITLE
Breaking build termination on latest Jenkins

### ISSUE THREAD

**Issue by hach-que** [jenkinsci/periodic-reincarnation-plugin#22]
I have the latest Jenkins, and at some point Periodic Reincarnation broke on a new version of Jenkins. Now when my builds finish, they end up in states like this, where they're technically finished, but still holding the slot on the slaves open:

![image](https://user-images.githubusercontent.com/504826/30697883-b7a16f40-9f23-11e7-9334-108deac0c178.png)

![image](https://user-images.githubusercontent.com/504826/30697895-c4a331e2-9f23-11e7-9ea2-7942748d1f5f.png)

![image](https://user-images.githubusercontent.com/504826/30697920-d34d266c-9f23-11e7-9e6d-18a1da810a46.png)
^ The spinner is still going, even though the main build page doesn't show the build as "building".

The logs show this error:

```
A thread (PeriodicReincarnation thread/2813) died unexpectedly due to an uncaught exception, this may leave your Jenkins in a bad way and is usually indicative of a bug in the code.
java.lang.ClassCastException: org.jenkinsci.plugins.workflow.job.WorkflowJob cannot be cast to hudson.model.AbstractProject
	at org.jenkinsci.plugins.periodicreincarnation.PeriodicReincarnation.checkProject(PeriodicReincarnation.java:325)
	at org.jenkinsci.plugins.periodicreincarnation.PeriodicReincarnation.checkFolder(PeriodicReincarnation.java:353)
	at org.jenkinsci.plugins.periodicreincarnation.PeriodicReincarnation.addProjectsFoundByPeriodicTriggerHit(PeriodicReincarnation.java:315)
	at org.jenkinsci.plugins.periodicreincarnation.PeriodicReincarnation.execute(PeriodicReincarnation.java:115)
	at hudson.model.AsyncPeriodicWork$1.run(AsyncPeriodicWork.java:101)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by hach-que** [comment]
Confirmed uninstalling 1.11 version of this plugin resolves the hanging builds issue.

**Comment by hach-que** [comment]
Downgrading to 1.10 also resolves the issue, so it looks like this is just a problem with version 1.11.

**Comment by Jochen-A-Fuerbacher** [comment]
Thanks for reporting this issue. I fixed the error of the log you provided. We could not reproduce the problem with the "hanging builds". So I'm not quite sure, if this fix resolves your issue. I'll release a new version soon. Please comment here if the problem is resolved or not.

**Comment by hach-que** [comment]
That seems to have fixed the issue for me.

**Comment by krulls** [comment]
relates to https://issues.jenkins-ci.org/browse/JENKINS-46419

**Comment by hach-que** [comment]
I'm seeing this issue again on 1.12, now with this stack trace:

```
A thread (PeriodicReincarnation thread/40206) died unexpectedly due to an uncaught exception, this may leave your Jenkins in a bad way and is usually indicative of a bug in the code.
java.lang.ClassCastException: org.jenkinsci.plugins.workflow.job.WorkflowJob cannot be cast to hudson.model.AbstractProject
	at org.jenkinsci.plugins.periodicreincarnation.PeriodicReincarnation.checkProject(PeriodicReincarnation.java:400)
	at org.jenkinsci.plugins.periodicreincarnation.PeriodicReincarnation.checkFolder(PeriodicReincarnation.java:393)
	at org.jenkinsci.plugins.periodicreincarnation.PeriodicReincarnation.addUnchangedProjects(PeriodicReincarnation.java:374)
	at org.jenkinsci.plugins.periodicreincarnation.PeriodicReincarnation.execute(PeriodicReincarnation.java:117)
	at hudson.model.AsyncPeriodicWork$1.run(AsyncPeriodicWork.java:101)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by hach-que** [comment]
(Again it's causing a build to not terminate correctly)

**Comment by Jochen-A-Fuerbacher** [comment]
Thanks for reporting. Should be fixed in 1.12.1. Again, please report any issue. Thank you.
